lib/logwriter.c

```diff
@@ -1203,17 +1206,15 @@ log_writer_write_message(LogWriter *self, LogMessage *msg, LogPathOptions *path_
 
       self->partial_write = (status == LPS_PARTIAL);
 
-      if (consumed)
-        log_writer_realloc_line_buffer(self);
+      log_writer_realloc_line_buffer(self, consumed);
```

Comment:
This does not look like an equivalent transformation to me.
What's the idea behind this patch?

Reply:
The idea was that the `freeing` was outside the other buffer manipulations and I wanted them to be close to each other (on the call site -> inside the function).

The `consumed` boolean is responsible for deciding whether a specific buffer has been passed yet or not. If it has been passed, then it should not be freed, because the reference is needed (there is something that refers to it).
If it has not been passed (nothing else refers to it), then the old buffer must be freed.

Basically I moved the `g_free` call into the function body.